Test/ITKImageProcessingReaderTest.cpp

```diff
@@ -0,0 +1,322 @@
+    if (!reader)
+    {
+      return EXIT_FAILURE;
+    }
+    bool propertySet = false;
+
+    const QString containerName = "TestContainer";
+    DataContainer::Pointer inputContainer = DataContainer::New(containerName);
+    DataContainerArray::Pointer inputContainerArray = DataContainerArray::New();
+    inputContainerArray->addDataContainer(inputContainer);
+
+    reader->setDataContainerArray(inputContainerArray);
+    propertySet = reader->setProperty("DataContainerName", containerName);
+    DREAM3D_REQUIRE_EQUAL(propertySet, true);
+
+    propertySet = reader->setProperty("FileName", file);
+    DREAM3D_REQUIRE_EQUAL(propertySet, true);
+
+    reader->execute();
+    DREAM3D_REQUIRED(reader->getErrorCondition(), >= , 0);
+    DREAM3D_REQUIRED(reader->getWarningCondition(), >= , 0);
+
+    // Compare read data
+    DataContainerArray::Pointer containerArray = reader->getDataContainerArray();
+    DREAM3D_REQUIRE_NE(containerArray.get(), 0);
+    DREAM3D_REQUIRE(containerArray->getDataContainerNames().contains(containerName));
+
+    DataContainer::Pointer container = containerArray->getDataContainer(containerName);
+    DREAM3D_REQUIRE_NE(container.get(), 0);
+
+    IGeometry::Pointer geometry = container->getGeometry();
+    DREAM3D_REQUIRE_EQUAL(geometry->getGeometryTypeAsString(), "ImageGeometry");
+    ImageGeom::Pointer imageGeometry = std::dynamic_pointer_cast<ImageGeom>(geometry);
+    DREAM3D_REQUIRE_NE(imageGeometry.get(), 0);
+
+    float tol = 1e-6;
+    float resolution[3];
+    imageGeometry->getResolution(resolution[0], resolution[1], resolution[2]);
+    float origin[3];
+    imageGeometry->getOrigin(origin[0], origin[1], origin[2]);
+    size_t dimensions[3];
+    imageGeometry->getDimensions(dimensions[0], dimensions[1], dimensions[2]);
+    for (int i = 0; i < 3; i++)
+    {
+      //float fspacing = expectedImage->GetSpacing()[i];
+      float imageSpacing = 1.0; // SCIFIO doesn't seem to take spacing into consideration
+      DREAM3D_COMPARE_FLOATS(&resolution[i], &imageSpacing, tol);
+
+      //float forigin = expectedImage->GetOrigin()[i];
+      float imageOrigin = 0.0; // SCIFIO doesn't seem to take origin into consideration
+      DREAM3D_COMPARE_FLOATS(&origin[i], &imageOrigin, tol);
+
+      size_t imageDimension = expectedImage->GetLargestPossibleRegion().GetSize()[i];
+      DREAM3D_ASSERT(dimensions[i] == imageDimension);
+    }
+
+    const QString matrixName = "ImageData";
+    AttributeMatrix::Pointer attributeMatrix = container->getAttributeMatrix(matrixName);
+    DREAM3D_REQUIRE_NE(attributeMatrix.get(), 0);
+    IDataArray::Pointer dataArray = attributeMatrix->getAttributeArray("CellData");
+
+    // ANYTHING BELOW HERE FAILS WHEN IT SHOULD NOT
+    // vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
+    DREAM3D_REQUIRE_EQUAL(static_cast<PixelType*>(dataArray->getVoidPointer(0)), expectedImage->GetBufferPointer());
```

Comment:
This is never going to be true:
*dataArray points to some memory allocated by the itkImageFileReader
*expectedImage was created before the image is read
The two memory allocations are different, and therefore the pointers point to different addresses. Only the values contained in the arrays should be compared, as it is done below.